src/scratchtocatrobat/converter/test_converter.py

```diff
@@ -237,34 +237,70 @@ def get_sprite_with_soundinfo(self, soundinfo_name):
         self.assertNotEqual(catr_script.getBroadcastMessage(), broadcast_message)
         self.assertEqual(catr_script.getBroadcastMessage(), broadcast_message.lower())
 
     # whenKeyPressed
     def test_can_convert_keypressed_script(self):
         scratch_script = scratch.Script([30, 355, [["whenKeyPressed", "space"], ["changeGraphicEffect:by:", "color", 25]]])
-        catr_script = self.block_converter._catrobat_script_from(scratch_script, DUMMY_CATR_SPRITE)
+        catr_script = self.block_converter._catrobat_script_from(scratch_script, DUMMY_CATR_SPRITE, self.test_project)
         # KeyPressed-scripts are represented with broadcast-scripts with a special key-message
         self.assertTrue(isinstance(catr_script, catbase.BroadcastScript))
         self.assertEqual(converter._key_to_broadcast_message("space"), catr_script.getBroadcastMessage())
 
     # whenClicked
     def test_can_convert_whenclicked_script(self):
         scratch_script = scratch.Script([30, 355, [["whenClicked"], ["changeGraphicEffect:by:", "color", 25]]])
-        catr_script = self.block_converter._catrobat_script_from(scratch_script, DUMMY_CATR_SPRITE)
+        catr_script = self.block_converter._catrobat_script_from(scratch_script, DUMMY_CATR_SPRITE, self.test_project)
         self.assertTrue(isinstance(catr_script, catbase.WhenScript))
         self.assertEqual('Tapped', catr_script.getAction())
 
+    #whenSensorGreaterThan
+    def test_can_convert_when_loudness_greater_than_script_with_formula(self):
+        scratch_script= scratch.Script([30, 355, [["whenSensorGreaterThan", "loudness", ["+", 2, 1]], ["say:", "Hello!"]]])
+        catr_script = self.block_converter._catrobat_script_from(scratch_script, DUMMY_CATR_SPRITE, self.test_project)
+        
+        assert isinstance(catr_script, catbase.WhenConditionScript)
+        formula = catr_script.formulaMap[catbricks.Brick.BrickField.IF_CONDITION] #@UndefinedVariable
+        assert isinstance(formula, catformula.Formula)
+        
+        assert formula.formulaTree.value == str(catformula.Operators.GREATER_THAN)
+        assert formula.formulaTree.leftChild.value == str(catformula.Sensors.LOUDNESS)
+        assert formula.formulaTree.rightChild.value == str(catformula.Operators.PLUS)
+        assert formula.formulaTree.rightChild.leftChild.value == "2"
+        assert formula.formulaTree.rightChild.rightChild.value == "1"
+
+    #whenSensorGreaterThan
+    def test_can_convert_when_timer_greater_than_script_with_formula(self):
+        scratch_script= scratch.Script([30, 355, [["whenSensorGreaterThan", "timer", ["+", 2, 1]], ["say:", "Hello!"]]])
+        catr_script = self.block_converter._catrobat_script_from(scratch_script, DUMMY_CATR_SPRITE, self.test_project)
+        
+        assert isinstance(catr_script, catbase.WhenConditionScript)
+        formula = catr_script.formulaMap[catbricks.Brick.BrickField.IF_CONDITION] #@UndefinedVariable
+        assert isinstance(formula, catformula.Formula)
```

Comment:
Actually, there is still one crucial thing missing in this test:
The timer workaround has to be tested as well.
In order to do, I'd suggest to wrap a sprite-object JSON code around the script and pass this to scratch.Object() constructor instead.
I assume this shouldn't be very tricky at all.
However, in case my assumption is wrong, please let me know.